### PR TITLE
#772: Check installation of R packages

### PR DESCRIPTION
**Pull request by tomuben**
Throw an error if installation of R package fails and run integration tests for R package in standard container

related to https://github.com/exasol/script-languages-release/issues/772

### CODE REVIEW

File: ext/scripts/install_scripts/install_via_r_remotes.pl [hunk 1]
```diff
@@ -54,13 +54,57 @@ =head1 SYNOPSIS
...
+}
+
+
+<<<<0>>>>
+';
+
+
 my @separators = ("\n");
-my @templates = ('install_version("<<<<0>>>>",NULL,repos="https://cloud.r-project.org", Ncpus=4)');
+my @templates = ('install_or_fail("<<<<0>>>>",NULL)');
```
Comment (tkilias) [inline ext/scripts/install_scripts/install_via_r_remotes.pl:101]: Maybe let's rename it to install template

File: ext/scripts/install_scripts/install_via_r_remotes.pl [hunk 2]
```diff
@@ -54,13 +54,57 @@ =head1 SYNOPSIS
...
+           if(catch){ print(w$message)
+             stop(paste("installation failed for:",package_name ))}}
+         )
+
+ }
+
+<<<<0>>>>
+';
+
+my $combining_template_validation = '
```
Comment (tkilias) [inline ext/scripts/install_scripts/install_via_r_remotes.pl:81]: Not sure if we need it, because we already do a library call during the install

Comment (tomuben) [reply ext/scripts/install_scripts/install_via_r_remotes.pl:81]: This function checks if the package is in `available.packages()`.
Point 2 in the [acceptance](https://github.com/exasol/script-languages-release/issues/772#issuecomment-1468173297) criteria of the related ticket.

File: ext/scripts/install_scripts/install_via_r_remotes.pl (hunk 1 above)
Comment (tomuben) [reply ext/scripts/install_scripts/install_via_r_remotes.pl]: Done

File: ext/scripts/install_scripts/install_via_r_remotes.pl [hunk 3]
```diff
@@ -74,13 +131,19 @@ sub replace_missing_version{
 }
 
 my @rendered_line_transformation_functions = (\&identity);
+my @rendered_line_transformation_functions_validation = (\&identity);
 if($with_versions and $allow_no_version){
     @rendered_line_transformation_functions = (\&replace_missing_version);
```
Comment (tkilias) [inline ext/scripts/install_scripts/install_via_r_remotes.pl:136]:
```suggestion
    @rendered_line_transformation_functions_install = (\&replace_missing_version);
```

File: ext/scripts/install_scripts/install_via_r_remotes.pl [hunk 4]
```diff
@@ -74,13 +131,19 @@ sub replace_missing_version{
...
+    @rendered_line_transformation_functions_validation = (\&replace_missing_version);
 }
 
 my $script = 
     package_mgmt_utils::generate_joined_and_transformed_string_from_file(
-        $file,$element_separator,$combining_template,\@templates,\@separators,\@rendered_line_transformation_functions);
+        $file,$element_separator,$combining_template_install,\@install_templates,\@separators,\@rendered_line_transformation_functions) .
```
Comment (tkilias) [inline ext/scripts/install_scripts/install_via_r_remotes.pl:142]:
```suggestion
        $file,$element_separator,$combining_template_install,\@install_templates,\@separators,\@rendered_line_transformation_functions_install) .
```

File: ext/scripts/install_scripts/install_via_r_remotes.pl [hunk 5]
```diff
@@ -74,13 +131,19 @@ sub replace_missing_version{
 }
 
 my @rendered_line_transformation_functions = (\&identity);
```
[the remark this answers is not in the document]
Comment (tomuben) [reply ext/scripts/install_scripts/install_via_r_remotes.pl:133]: Done

File: ext/scripts/install_scripts/install_via_r_remotes.pl (hunk 3 above)
Comment (tomuben) [reply ext/scripts/install_scripts/install_via_r_remotes.pl:136]: Done

File: ext/scripts/install_scripts/install_via_r_remotes.pl (hunk 4 above)
Comment (tomuben) [reply ext/scripts/install_scripts/install_via_r_remotes.pl:142]: Done

**Review by tkilias — APPROVED** [review]

**Pull request merged**